qa/rpc-tests/listtransactions.py

```diff
@@ -7,9 +7,21 @@
 # Exercise the listtransactions API
 
 from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
 
 from decimal import Decimal
 
+def count_array_matches(object_array, to_match):
+    num_matched = 0
+    for item in object_array:
+        all_match = True
+        for key,value in to_match.items():
+            if item[key] != value:
+                all_match = False
+        if all_match:
```

Comment:
```suggestion
        if all((item[key] == value for key,value in to_match.items())):
```
(As well as being more concise, this [short-cuts if a key/value does not match](https://docs.python.org/3/library/functions.html#all).)